java/code/src/com/suse/manager/reactor/messaging/RegistrationUtils.java

```diff
@@ -352,12 +353,12 @@ private static Set<Channel> findChannelsForProducts(Set<SUSEProduct> suseProduct
      * @param minionId the minionId
      * @param channels assigned channels
      * @param grains the system grains
+     * @param instanceFlavor public cloud instance flavor
      * @return true if the use is allowed
      */
-    public static boolean isAllowedOnPayg(SystemQuery systemQuery, String minionId,
-                                          Set<Channel> channels, ValueMap grains) {
-        boolean isPaygClient = grains.getOptionalAsBoolean("is_payg_instance").orElse(false);
-        if (isPaygClient) {
+    public static boolean isAllowedOnPayg(SystemQuery systemQuery, String minionId, Set<Channel> channels,
+                                          ValueMap grains, PublicCloudInstanceFlavor instanceFlavor) {
+        if (instanceFlavor.name() == "PAYG") {
```

Comment:
we can use the enum value here to compare `instanceFlavor == PublicCloudInstanceFlavor.PAYG`